server/apps/immich/src/controllers/user.controller.ts

```diff
@@ -106,4 +107,22 @@ export class UserController {
     res.header('Content-Type', 'image/jpeg');
     return new StreamableFile(readableStream);
   }
+
+  @Authenticated()
+  @Get('partner')
+  getAllPartners(@GetAuthUser() authUser: AuthUserDto): Promise<PartnerResponseDto[]> {
+    return this.service.getAllPartners(authUser);
+  }
+
+  @Authenticated()
+  @Post('partner/:id')
+  addPartner(@GetAuthUser() authUser: AuthUserDto, @Param() { id }: UUIDParamDto): Promise<void> {
+    return this.service.addPartner(authUser, id);
+  }
+
+  @Authenticated()
+  @Delete('partner/:id')
+  removePartner(@GetAuthUser() authUser: AuthUserDto, @Param() { id }: UUIDParamDto): Promise<void> {
+    return this.service.removePartner(authUser, id);
+  }
```

Comment:
This looks great, but it would be better IMO for this to be in a separate controller, `PartnerController` with the url `/partner/...` instead of `/user/partner/...`